sktime/forecasting/compose/_pipeline.py

```diff
@@ -154,7 +154,15 @@ def _get_inverse_transform(self, transformers, y, X=None, mode=None):
     @property
     def named_steps(self):
         """Map the steps to a dictionary."""
-        return dict(self.steps)
+        return dict(self._steps)
+
+    @property
+    def _steps(self):
+        return self._get_estimator_tuples(self.steps, clone_ests=False)
+
+    @_steps.setter
+    def _steps(self, value):
```

Comment:
Just out of curiosity question, I got lost a bit with inner steps. Where is this setter used?

Reply:
this is part of the fix in #2619, not the notebook PR.

It's required to write `_steps` write to `steps`, to ensure that `steps` and `_steps` are always in sync.

It's required for the `_HeterogeneousMetaEstimator._set_params` to work, since that targets `_steps` and not `steps`, and `_get_params` reads from `_steps` which is a standard formatted version of `steps`.

Reply:
Thanks for clarifying, I think I understand now